vcap_utils/vcap_utils/backends/backend_rpc_process.py

```diff
@@ -0,0 +1,204 @@
+        self._futures_lock = RLock()
+        self._futures: Dict[str, Future] = {}
+        """Keep track of ongoing requests in a dict of request_id: Future """
+
+        # Spin up the server process
+        self._process = multiprocessing.Process(
+            target=_rpc_server,
+            daemon=True,
+            name="RpcServer",
+            kwargs={
+                "outgoing": self._incoming,
+                "incoming": self._outgoing,
+                "shutdown": self._shutdown,
+                "backend_class": backend_class,
+                "num_workers": 100,
+                "args": args,
+                "kwargs": kwargs
+            })
+        self._process.start()
+        # Wait for success or failure of the Backend initialization
+        exception = self._incoming.get()
+        if exception:
+            raise exception
+
+        # Start the client thread
+        self._rpc_thread = threading.Thread(
+            target=self._rpc_client,
+            name="RpcClientThread")
+        self._rpc_thread.start()
+
+    @property
+    def workload(self) -> int:
+        # TODO: Use RPC to request the underlying Backends
+        #       'workload' implementation
+        with self._futures_lock:
+            return len(self._futures)
+
+    def _rpc_client(self):
+        """This is the dedicated thread for receiving and routing results
+        from the remote backend. It receives incoming RpcResponses and
+        routes results or exceptions to the relevant Future objects."""
+        while not self._shutdown.is_set() or self._incoming.qsize():
+            try:
+                response: RpcResponse = self._incoming.get(timeout=0.1)
+            except queue.Empty:
+                continue
+            with self._futures_lock:
+                future = self._futures.pop(response.request_id)
+            if response.exception:
+                future.set_exception(response.exception)
+            else:
+                future.set_result(response.result)
+
+    def _rpc_call(self, function: str, *args, **kwargs):
+        """Run a function on the remote backend.
+        :param function: The function to run on the remote backend
+        :param args: Arguments for the function
+        :param kwargs: Keyword arguments for the function
+        """
+        request = RpcRequest(
+            function=function,
+            args=args,
+            kwargs=kwargs,
+            request_id=uuid4().hex
+        )
+
+        future = Future()
+        with self._futures_lock:
+            self._futures[request.request_id] = future
+        self._outgoing.put(request)
+        return future.result()
+
+    def process_frame(self, *args, **kwargs) -> DETECTION_NODE_TYPE:
+        return self._rpc_call("process_frame", *args, **kwargs)
+
+    def distances(self, *args, **kwargs):
+        return self._rpc_call("distances", *args, **kwargs)
+
+    def batch_predict(self, *args, **kwargs):
```

Comment:
Type annotate return type?

Reply:
I feel weird about that, considering I don't know the return type, and it's explicitly NotImplemented. This method won't ever be implemented by a child class, either.

Reply:
`-> typing.NoReturn`

Reply:
https://www.python.org/dev/peps/pep-0484/#the-noreturn-type

Reply:
Nice, I added it.